lib/rewards/server/subscribers/world-drop-subscriber.js

```diff
@@ -0,0 +1,149 @@
+const { sc, Logger } = require('@reldens/utils');
+const { RewardsConst } = require('../../constants');
+
+
+class WorldDropSubscriber
+{
+    static async dropRewardItem(params)
+    {
+        let { rewardEventData, scene } = params;
+        let itemRewards = sc.get(rewardEventData, 'itemRewards', false);
+        if (!itemRewards) {
+            return;
+        }
+        let rewards = itemRewards.map((itemReward) => itemReward.reward);
+        let targetObjectBody = sc.get(rewardEventData.target_object, 'objectBody', false);
+        if (!targetObjectBody) {
+            return;
+        }
+        let closerWalkableNodes = this.getWalkableNodesAround(targetObjectBody);
+        let hasSpace = this.hasEnoughFreeNodesForRewards(rewards, closerWalkableNodes);
+
+        for (let reward of rewards) {
+            try {
+                reward['rewardPosition'] = this.getRewardPosition(closerWalkableNodes, hasSpace);
+                reward['hasSpace'] = hasSpace;
+                let tileIndex = this.getTileIndex(reward.rewardPosition, targetObjectBody.worldWidth);
+                reward['randomRewardId'] = `reward[${tileIndex}]-${sc.randomChars(8)}`;
+                reward['objectBody'] = await this.createRewardRoomObject(scene, targetObjectBody, reward);
+                if (!reward['objectBody']) {
+                    Logger.error('No object body for reward ' + reward.id + ' could be created');
+                    return;
+                }
+            } catch (e) {
+                Logger.error(e);
+            }
+        }
+        return rewards;
+    }
+
+    static getWalkableNodesAround(targetObjectBody)
+    {
+        let { currentCol: col, currentRow: row } = targetObjectBody;
+        let pathfinder = targetObjectBody.getPathFinder();
+        let nodes = [pathfinder.grid.getNodeAt(col, row)];
+        for (let i = -1; i <= 1; i++) {
+            for (let j = -1; j <= 1; j++) {
+                let node = pathfinder.grid.getNodeAt(col + i, row + j);
+                if (node.walkable) {
+                    nodes.push(node);
+                }
+            }
+        }
+        return nodes;
+    }
+
+    static hasEnoughFreeNodesForRewards(itemRewards, nodes)
+    {
+        return nodes.length >= itemRewards.length;
+    }
+
+    static getTileIndex(node, worldWidth)
+    {
+        return node.x * worldWidth + node.y;
+    }
+
+    static getRewardsInteractionDistanceConfig(scene)
+    {
+        // this.range = medio tile (configurable);
+        // battle_rewards/node_modules/@reldens/skills/lib/skill.js
+        // let interactionArea = new InteractionArea();
+        // interactionArea.setupInteractionArea(this.range, targetPosition.x, targetPosition.y);
+        // let interactionResult = interactionArea.isValidInteraction(ownerPosition.x, ownerPosition.y);
+
+
+        // @TODO - BETA - rewards.interactionField
+        return scene.config.get('server/rewards/actions/interactionsDistance') ||
```

Comment:
config.get can receive a second parameter which will be used as "default", and in most of the cases we can use the withoutLogs version so we don't get the message on the console:
```
return scene.config.getWithoutLogs(
    'server/rewards/actions/interactionsDistance',
    (scene.roomWorld.mapJson.tilewidth * 0.5)
);
```